src/components/Auth/Login/Login.react.js

```diff
@@ -256,7 +166,93 @@ class Login extends Component {
     }
   };
 
+  handleServeChange = event => {
+    const { checked } = this.state;
+
+    switch (event.target.value) {
+      case 'customServer':
+        const defaults = UserPreferencesStore.getPreferences();
+        this.setState({
+          checked: !checked,
+          serverUrl: defaults.StandardServer,
+          serverFieldError: false,
+          customServerMessage: '',
+        });
+        break;
+      case 'serverUrl':
+        const serverUrl = event.target.value;
+        const validServerUrl = new RegExp(
+          [
+            '(http|ftp|https)://[w-]+(.[w-]+)+([',
+            'w.,@?^=%&amp;:~+#-]*[w@?^=%&amp;~+#-])?',
+          ].join(''),
+          'i',
+        ).test(serverUrl);
+        const serverFieldError = !(serverUrl && validServerUrl);
+        this.setState({
+          serverUrl,
+          serverFieldError,
+          customServerMessage: serverFieldError ? 'Enter a valid URL' : '',
+        });
+        break;
+      default:
+        break;
+    }
+  };
+
+  handleTextFieldChange = event => {
+    let email = '';
+    let password = '';
+
+    switch (event.target.name) {
+      case 'email':
+        email = event.target.value.trim();
+        const validEmail = new RegExp(
+          '^[A-Z0-9._%+-]+@[A-Z0-9.-]+.[A-Z]{2,4}$',
+          'i',
+        ).test(email);
+
+        this.setState({
+          email,
+          emailError: !validEmail || email === '',
+          emailErrorMessage:
+            !validEmail || email === '' ? 'Enter a valid Email Address' : '',
+        });
+        break;
+      case 'password':
+        password = event.target.value.trim();
+        this.setState({
+          password,
+          passwordError: password === '',
+          passwordErrorMessage: password === '' ? 'Enter a valid password' : '',
+        });
+        break;
+      default:
+        break;
+    }
+  };
+
   render() {
+    const {
+      email,
+      password,
+      loading,
+      showDialog,
+      message,
+      emailError,
+      passwordError,
+      serverFieldError,
+      emailErrorMessage,
+      passwordErrorMessage,
+    } = this.state;
+
+    const isValid =
+      serverFieldError ||
+      emailError ||
+      passwordError ||
+      email === '' ||
+      password === '';
+
     const styles = {
```

Comment:
How about we move the styles object outside the render function? :thinking:

Reply:
I was actually hoping that we would completely get rid of these when we integrate styled-components. But i'll just move these below the component for now.